core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -312,12 +289,13 @@ public void run() {
 
     @Override
     public void closed(Status status, Metadata trailers) {
-      if (status.getCode() == Status.Code.CANCELLED && deadlineNanoTime != null) {
+      Long timeoutMicros = getRemainingTimeoutMicros();
+      if (status.getCode() == Status.Code.CANCELLED && timeoutMicros != null) {
         // When the server's deadline expires, it can only reset the stream with CANCEL and no
         // description. Since our timer may be delayed in firing, we double-check the deadline and
         // turn the failure into the likely more helpful DEADLINE_EXCEEDED status. This is always
         // safe, but we avoid wasting resources getting the nanoTime() when unnecessary.
```

Comment:
Should this comment be updated in light of this change?  Specifically "we avoid wasting resources getting the nanoTime() "